packages/meditrak-server/src/tests/routes/GETSurveyResponses/assertSurveyResponsesPermission.test.js

```diff
@@ -0,0 +1,144 @@
+      },
+      {
+        id: generateTestId(),
+        code: 'TEST_SURVEY_2',
+        name: 'Test Survey 2',
+        permission_group_id: donorPermissionGroup.id,
+      },
+    ]);
+
+    const surveyResponseModels = await buildAndInsertSurveyResponses(models, [
+      {
+        id: generateTestId(),
+        surveyCode: 'TEST_SURVEY_1',
+        entityCode: vanuatuEntity.code,
+        submission_time: '2020-01-31T09:00:00Z',
+        answers: [],
+      },
+      {
+        id: laosAdminResponseId,
+        surveyCode: 'TEST_SURVEY_1',
+        entityCode: laosEntity.code,
+        submission_time: '2020-01-31T09:00:00Z',
+        answers: [],
+      },
+      {
+        id: generateTestId(),
+        surveyCode: 'TEST_SURVEY_2',
+        entityCode: vanuatuEntity.code,
+        submission_time: '2020-01-31T09:00:00Z',
+        answers: [],
+      },
+      {
+        id: generateTestId(),
+        surveyCode: 'TEST_SURVEY_2',
+        entityCode: laosEntity.code,
+        submission_time: '2020-01-31T09:00:00Z',
+        answers: [],
+      },
+    ]);
+    surveyResponses = surveyResponseModels.map(srm => srm.surveyResponse);
+  });
+
+  describe('filterSurveyResponsesByPermissions()', async () => {
+    it('Should filter any survey responses that users do not have access to to the survey of', async () => {
+      //Remove the permission of VU to have insufficient permissions to some surveys
+      const policy = {
+        // VU: ['Admin', 'Donor'],
+        LA: ['Admin'],
+      };
+      const accessPolicy = new AccessPolicy(policy);
+      const result = await filterSurveyResponsesByPermissions(
+        accessPolicy,
+        surveyResponses,
+        models,
+      );
+
+      // Should have dropped all but [laos/Admin] survey response
+      expect(result.length).to.equal(1);
+      expect(result[0].id).to.equal(laosAdminResponseId);
+    });
+  });
+
+  describe('assertSurveyResponsePermissions()', async () => {
+    it('Sufficient permissions: Should return true if users have access to the survey permission group in the entity country', async () => {
+      const accessPolicy = new AccessPolicy(DEFAULT_POLICY);
+      const result = await assertSurveyResponsePermissions(
+        accessPolicy,
+        surveyResponses[0],
+        models,
+      );
+
+      expect(result).to.true;
+    });
+
+    it('Insufficient permissions: Should throw an exception if users do not have access to the survey permission group in the entity country', async () => {
+      // Remove all default permissions to always have insufficient permissions
+      const policy = {
+        // VU: ['Admin', 'Donor'],
+        // LA: ['Admin', 'Donor'],
+        DL: ['Admin'],
+      };
+      const accessPolicy = new AccessPolicy(policy);
+
+      // Should only have permission for [Laos/Admin]
+      expect(() => assertSurveyResponsePermissions(accessPolicy, surveyResponses[0], models)).to
+        .throw;
```

Comment:
```suggestion
      expect(() => assertSurveyResponsePermissions(accessPolicy, models, surveyResponses[0], )).to
        .throw;
```

I think the parameters are in the wrong order? `assertSurveyResponsePermissions` function signature is
`export const assertSurveyResponsePermissions = async (accessPolicy, models, surveyResponse)`

Reply:
You're totally right. I think I had some local changes while I was first running these. I've fixed it up.